### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by Othniel2471**
### In this PR @souad and I :
- Added Login page and registration page.
- Added Food list.
- > displays a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- > leads to a form that allows users to add new food.
- Added Recipes list.
- > displays a list of recipes created by the logged-in user as in the wireframe.
- > leads to recipe details.
- > If the user is the owner of the recipe, allows the user to delete it.
- Added Public recipe list.
- > displays a list of all public recipes ordered by newest as in the wireframe
- > leads to recipe details.
- Added Recipe details.
-  Added General shopping list view.
- Added Navigation.
- Add unit tests and integration tests.

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
+    authorize! :destroy, @food
+    @food.destroy
+    redirect_to foods_url, notice: 'Food was successfully destroyed.'
+  rescue ActiveRecord::InvalidForeignKey
+    redirect_to foods_url, alert: 'This food item cannot be deleted because it is associated with other records.'
+  end
```
Comment (VICTOR-CHUKWU) [inline app/controllers/foods_controller.rb:57]: - [ ] very excellent error handling done here, good work guys :+1:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Othniel2471  and @souad988 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] App works correctly
- [ ] error handling and flash message done superbly and with nice UI
- [ ] devise installed and used correctly
- [ ] app looks similar to given wireframe
- [ ] good commit history between both team members

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/foods_controller.rb (hunk 1 above)
Comment (Othniel2471) [reply app/controllers/foods_controller.rb:57]: thank you very much

**Comment by Othniel2471** [comment]
Thank you

**Pull request merged**